src/autopas/containers/linkedCells/traversals/C04Traversal.h

```diff
@@ -0,0 +1,233 @@
+   * @param cutoff Cutoff radius.
+   * @param cellLength cell length.
+   */
+  C04Traversal(const std::array<unsigned long, 3> &dims, PairwiseFunctor *pairwiseFunctor, const double cutoff = 1.0,
+               const std::array<double, 3> &cellLength = {1.0, 1.0, 1.0})
+      : C08BasedTraversal<ParticleCell, PairwiseFunctor, DataLayout, useNewton3>(dims, pairwiseFunctor, cutoff,
+                                                                                 cellLength),
+        _cellHandler(pairwiseFunctor, this->_cellsPerDimension, cutoff, cellLength, this->_overlap) {
+    computeOffsets32Pack();
+  }
+
+  /**
+   * @copydoc LinkedCellTraversalInterface::traverseCellPairs()
+   */
+  void traverseCellPairs(std::vector<ParticleCell> &cells) override;
+
+  TraversalOption getTraversalType() const override { return TraversalOption::c04; }
+
+  /**
+   * C04 traversals are always usable.
+   * @return
+   */
+  bool isApplicable() const override {
+    int nDevices = 0;
+#if defined(AUTOPAS_CUDA)
+    cudaGetDeviceCount(&nDevices);
+#endif
+    if (DataLayout == DataLayoutOption::cuda)
+      return nDevices > 0;
+    else
+      return true;
+  }
+
+ private:
+  void traverseCellPairsBackend(std::vector<ParticleCell> &cells, const std::array<long, 3> &start,
+                                const std::array<long, 3> &end);
+
+  void traverseSingleColor(std::vector<ParticleCell> &cells, int color, const std::array<long, 3> &start,
+                           const std::array<long, 3> &end);
+
+  void processBasePack32(std::vector<ParticleCell> &cells, const std::array<long, 3> &base3DIndex,
+                         const std::array<long, 3> &start, const std::array<long, 3> &end);
+
+  void computeOffsets32Pack();
+
+  long parity(long x, long y, long z) const { return (x + y + z + 24) % 8; }
+
+  std::array<std::array<long, 3>, 32> _cellOffsets32Pack;
+
+  C08CellHandler<ParticleCell, PairwiseFunctor, DataLayout, useNewton3> _cellHandler;
+};
+
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption DataLayout, bool useNewton3>
+inline void C04Traversal<ParticleCell, PairwiseFunctor, DataLayout, useNewton3>::traverseCellPairs(
+    std::vector<ParticleCell> &cells) {
+  std::array<long, 3> start, end;
+  for (int d = 0; d < 3; ++d) {
+    start[d] = 0l;
+    end[d] = static_cast<long>(this->_cellsPerDimension[d]) - 1;
+  }
+  traverseCellPairsBackend(cells, start, end);
+}
+
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption DataLayout, bool useNewton3>
+void C04Traversal<ParticleCell, PairwiseFunctor, DataLayout, useNewton3>::computeOffsets32Pack() {
+  using std::make_pair;
+  using utils::ThreeDimensionalMapping::threeToOneD;
+
+  int i = 0;
+  long z = 0l;
+  _cellOffsets32Pack[i++] = {1l, 1l, z};
+  _cellOffsets32Pack[i++] = {1l, 2l, z};
+  _cellOffsets32Pack[i++] = {2l, 1l, z};
+  _cellOffsets32Pack[i++] = {2l, 2l, z};
+
+  /* z = 1ul; z = 2ul */
```

Comment:
Either remove or change to:
```suggestion
  // z = 1ul; z = 2ul 
```
otherwise this might disrupt doxygen